unified_planning/io/python_writer.py

```diff
@@ -0,0 +1,332 @@
+        
+        for t, gl in self.problem.timed_goals().items(): # add timed goals
+            for g in gl:
+                out.write(f'problem.add_timed_goal(timing={_convert_timing(t)}, goal={converter.convert(g)})\n')
+        
+        for i, gl in self.problem.maintain_goals().items(): # add maintain goals
+            for g in gl:
+                out.write(f'problem.add_maintain_goal(interval={_convert_interval(i)}, goal={converter.convert(g)})\n')
+        
+        for g in self.problem.goals(): # add goals
+            out.write(f'problem.add_goal(goal={converter.convert(g)})\n')
+
+        for qm in self.problem.quality_metrics():
+            out.write('problem.add_quality_metric(')
+            if isinstance(qm, up.model.metrics.MinimizeActionCosts):
+                out.write('up.model.metrics.MinimizeActionCosts()')
+            elif isinstance(qm, up.model.metrics.MinimizeExpressionOnFinalState):
+                out.write(f'up.model.metrics.MinimizeExpressionOnFinalState({converter.convert(qm.expression)})')
+            elif isinstance(qm, up.model.metrics.MaximizeExpressionOnFinalState):
+                out.write(f'up.model.metrics.MaximizeExpressionOnFinalState({converter.convert(qm.expression)})')
+            else:
+                raise NotImplementedError
+            out.write(')\n')
+
+    def print_domain(self):
+        '''Prints to std output the PDDL domain.'''
+        self._write_problem_code(sys.stdout)
+
+    def write_problem_code(self) -> str:
+        '''Returns the PDDL domain.'''
+        out = StringIO()
+        self._write_problem_code(out)
+        return out.getvalue()
+
+    def write_domain(self, filename: str):
+        '''Dumps to file the PDDL domain.'''
+        with open(filename, 'w') as f:
+            self._write_problem_code(f)
+
+
+def _print_python_type(type: 'up.model.types.Type') -> str:
+    '''This method takes a type and returns how to use it from command line.
+    For the user_types it assumes they have already been created and just refers to them.'''
+    if type.is_user_type():
+        return f'type_{cast(_UserType, type).name()}'
+    elif type.is_bool_type():
+        return 'BoolType()'
+    elif type.is_int_type():
+        return f'IntType({str(cast(_IntType, type).lower_bound())}, {str(cast(_IntType, type).upper_bound())})'
+    elif type.is_real_type():
+        return f'RealType({str(cast(_RealType, type).lower_bound())}, {str(cast(_RealType, type).upper_bound())})'
+    else:
+        raise NotImplementedError
+
+def _convert_timing(timing: up.model.Timing) -> str:
+    bound: str = f'{str(timing.bound())}'
+    if isinstance(timing.bound(), Fraction):
+        bound = f'Fraction({str(timing.bound().numerator)}, {str(timing.bound().denomiantor)})'
+    if timing.is_from_start():
+        return f'StartTiming({bound})'
```

Comment:
I think you need to import these explicitly or use the full name `up.model.timing.StartTiming`